src/main/java/com/ksabbak/javaserver/app/controller/CatFormDataController.java

```diff
@@ -0,0 +1,58 @@
+package com.ksabbak.javaserver.app.controller;
+
+import com.ksabbak.javaserver.server.Response;
+import com.ksabbak.javaserver.server.StatusCode;
+import com.ksabbak.javaserver.storage.Persistable;
+
+import java.util.Map;
+
+public class CatFormDataController extends Controller {
```

Comment:
Even though there are cob spec tests, it's still best practice to be unit testing your controllers. Acceptance tests are not a replacement for unit tests, but are there for different types of feedback.

Completing the `Response`/`ResponseFormatter` refactor would definitely make it easier to unit test the controllers. Your tests would look something like:

```java
@Test
public void exampleTest() {
  Request request = new Request();
  FooController controller = new FooController();

  Response response = controller.get(request);

  assertEqual(200, response.getStatusCode());
  assertEqual("Some body", response.getBody());
  assertEqual("Header value", response.getHeader("SomeHeader"));
}
```

Reply:
Cool, did that still haven't  fixed Response though. :/